### ISSUE TITLE
Add relevance score to search results view

### ISSUE THREAD

**Issue by claude-cp** [keepassxreboot/keepassxc#7877]
# Proposition:

The first result should be the one that has the closest matching title (wildcards left and right).

# Summary

My search experience is bad right now because of questionable order of the results:
A lot of times I get say 20 results and have to keep looking for the one I want, which ends up being somewhere in the middle. 

The results are being shown in the order of the groups that they belong to (which looks like it can vary in its own turn). 
This is very un-intuitive behaviour, from a human perspective.

The behaviour described is present in both KeePassXC on linux, as well as in KeePassXC on Android - and ideally this should get handled in both places (I'm not sure if it is the exact same code that's being compiled for both platforms or not, hence mentioning just to be sure).

(**Edit**: I've trimmed down my comment because "searching in title only" has been relatively satisfactorily answered by @droidmonkey)

**Comment by droidmonkey** [comment]
Use `t:titletext` to search just the title. Example: `t:gmail`

Results sorted by "relevance" to search terms would be nice to have.

**Comment by relikd** [comment]
+1 for this feature. The search should be smart enough on its own to find more relevant entries on top instead of manually writing `t:` before the search term. Also, if there were a setting to only search title, that would also not suffice as I search by title 95% of the time but I wouldnt want to switch settings for the remaining 5%.

**Comment by droidmonkey** [comment]
"Smart enough" doesn't exist. Where you think Title is the most important, the next guy thinks url is. Hence why we introduced search field prefixes so you can hone in on what you care about. 

This is also lots of code compared to you typing t:

**Comment by relikd** [comment]
fair enough, but it is not me typing "t:" for a single search, I would want it for all my searches. And always typing "t:" before I can actually search is very inconvenient. My best suggestion so far would be to either make configurable what fields are search / or in what order. Or: have a default template search string (which would be the least development effort). e.g. "t:{}" and it will always prepend my search with "t:". I am not sure if I like either approach but I thought I just throw in my ideas if it helps anything.

**Comment by droidmonkey** [comment]
This request would interest you as well: https://github.com/keepassxreboot/keepassxc/issues/11072

**Comment by Cito** [comment]
I'm pretty sure it would be possible to find a simple ranking function that works for most people (is "smart enough"). Something like title first, then URL, then notes, then name and password, then the rest, and rank them higher if multiple of these fields match. Maybe rank higher if matching with word boundaries or upper/lower case. Also, rank all non-expired items higher than expired ones. And on equal rank, order by data of latest change descending. That would certainly yield more useful results in 99% of the cases than the current solution.

I understand that this is not done because it requires extra programming efforts that nobody pays for, but not because there is no exact definition of "smart enough".

**Comment by droidmonkey** [comment]
I let copilot give this a go, it did a pretty good job! Probably not entirely efficient, but this works:

<img width="1725" height="519" alt="Image" src="https://github.com/user-attachments/assets/d0414083-4fe0-4f03-baba-3c15ddfd9f28" />

You can give it a try in the PR linked above.
